rs/messaging/src/routing/stream_handler.rs

```diff
@@ -695,6 +727,13 @@ impl StreamHandlerImpl {
                     stream.signals_end(),
                     stream_index
                 );
+                let Ok(msg) = msg.try_into() else {
```

Comment:
The `try_into` might not be ideal here as it implicitly assumes that the only reason why this can fail is that the message is a blocker. Would some kind of match work here, e.g., on the concrete error we get back?